src/swarm/neo/client/mixins/SuspendableRequestCore.d

```diff
@@ -0,0 +1,1165 @@
+            as appropriate. (Setting the same value twice in a row has no
+            effect.)
+
+            Params:
+                ready = value to set for readiness flag
+
+        ***********************************************************************/
+
+        public void opAssign ( bool ready )
+        {
+            if ( ready )
+            {
+                if ( !this.ready_for_state_change )
+                {
+                    this.ready_for_state_change = true;
+                    this.suspendable_control.ready_for_state_change++;
+                }
+            }
+            else
+            {
+                if ( this.ready_for_state_change )
+                {
+                    assert(this.suspendable_control.ready_for_state_change > 0);
+                    this.ready_for_state_change = false;
+                    this.suspendable_control.ready_for_state_change--;
+                }
+            }
+        }
+    }
+
+    /// All-nodes initialiser used internally.
+    private AllNodesRequestInitialiser!(Request, FillPayload, HandleStatusCode)
+        all_nodes_initialiser;
+
+    /// State change readiness tracker instance.
+    private StateChangeReadinessTracker ready_for_state_change;
+
+    /***************************************************************************
+
+        Performs the logic of request initialisation. Called by
+        AllNodesRequestHandler.
+
+        Returns:
+            true if initialisation succeeded for this request-on-conn, false to
+            abort handling of the request on this connection
+
+    ***************************************************************************/
+
+    public bool initialise ( )
+    {
+        this.ready_for_state_change.suspendable_control =
+            &this.all_nodes_initialiser.context.shared_working.
+                suspendable_control;
+
+        auto suspendable_control =
+            &this.all_nodes_initialiser.context.shared_working.suspendable_control;
+
+        // If the desired state is already Stopped, simply abort initialisation
+        if ( suspendable_control.desired_state ==
+            suspendable_control.desired_state.Stopped )
+            return false;
+
+        // Perform standard all-nodes initialisation
+        if ( !this.all_nodes_initialiser.initialise() )
+            return false;
+
+        this.ready_for_state_change = true;
+
+        return true;
+    }
+
+    /***************************************************************************
+
+        Called by AllNodesRequest to reset any state required for initialisation
+        to begin again. Decrements the
+        SuspendableRequestSharedWorkingData.ready_for_state_change counter, if
+        it has been incremented for this request-on-conn.
+
+    ***************************************************************************/
+
+    public void reset ( )
+    {
+        this.ready_for_state_change = false;
```

Comment:
`this.all_nodes_initialiser.reset()` for future-proofness (as it is empty right now, might not be in the future)?